src/asfvideo.cpp

```diff
@@ -248,405 +238,273 @@ void AsfVideo::readMetadata() {
-  for (size_t& i : length) {
-    byte buf[2];
-    io_->read(buf, 2);
-    if (io_->error() || io_->eof())
-      throw Error(ErrorCode::kerFailedToReadImageData);
-    i = getUShort(buf, littleEndian);
+void AsfVideo::extendedStreamProperties() {
+  xmpData()["Xmp.video.StartTimecode"] = readQWORDTag();  // Start Time
+  xmpData()["Xmp.video.EndTimecode"] = readWORDTag();     // End Time
+
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Data Bitrate
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Buffer Size
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Initial Buffer Fullness
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Alternate Data Bitrate
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Alternate Buffer Size
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Alternate Initial Buffer Fullness
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Maximum Object Size
+  io_->seek(io_->tell() + DWORD, BasicIo::beg);  // ignore Flags Buffer Size
+  io_->seek(io_->tell() + WORD, BasicIo::beg);   // ignore Flags Stream Number
+  io_->seek(io_->tell() + WORD, BasicIo::beg);   // ignore Stream Language ID Index
+
+  xmpData()["Xmp.video.FrameRate"] = readWORDTag();  // Average Time Per Frame
+  uint16_t stream_name_count = readWORDTag();
+  uint16_t payload_ext_sys_count = readWORDTag();
+
+  for (uint16_t i = 0; i < stream_name_count; i++) {
+    io_->seek(io_->tell() + WORD, BasicIo::beg);  // ignore Language ID Index
+    uint16_t stream_length = readWORDTag();
+    if (stream_length)
+      io_->seek(io_->tell() + stream_length, BasicIo::beg);  // ignore Stream name
   }
-  for (int i = 0; i < 5; ++i) {
-    DataBuf buf(length[i]);
-    std::memset(buf.data(), 0x0, buf.size());
-    io_->read(buf.data(), length[i]);
-    if (io_->error() || io_->eof())
-      throw Error(ErrorCode::kerFailedToReadImageData);
-    const TagDetails* td = find(contentDescriptionTags, i);
-    assert(td);
-    std::string str(reinterpret_cast<const char*>(buf.data()), length[i]);
-    if (convertStringCharset(str, "UCS-2LE", "UTF-8")) {
-      xmpData()[td->label_] = str;
-    } else {
-      xmpData()[td->label_] = Util::toString16(buf);
-    }
+
+  for (uint16_t i = 0; i < payload_ext_sys_count; i++) {
+    io_->seek(io_->tell() + GUID, BasicIo::beg);  // ignore Extension System ID
+    io_->seek(io_->tell() + WORD, BasicIo::beg);  // ignore Extension Data Size
+    uint16_t ext_sys_info_length = readWORDTag();
+    if (ext_sys_info_length)
+      io_->seek(io_->tell() + ext_sys_info_length, BasicIo::beg);  // ignore Extension System Info
   }
-  if (io_->seek(pos + size, BasicIo::beg))
-    throw Error(ErrorCode::kerFailedToReadImageData);
-}  // AsfVideo::contentDescription
+}  // AsfVideo::extendedStreamProperties
 
 void AsfVideo::streamProperties() {
-  DataBuf buf(20);
-  byte guidBuf[GUI_SIZE];
-  int stream = 0;
+  byte streamTypedBuf[GUID];
+  io_->read(streamTypedBuf,GUID);
```

Comment:
clang-format-lint-action wants a space here
```suggestion
  io_->read(streamTypedBuf, GUID);
```

Reply:
Thank you @benmccann ,
Done here [b8f800e](https://github.com/Exiv2/exiv2/pull/2458/commits/b8f800ee4d784b179a934a2f38c7ddcd6762e1bc)